### PR TITLE
feat: upgrade to Ape 0.8

### PR DESCRIPTION
**Pull request by antazoey**

### CODE REVIEW

**Comment by antazoey** [comment]
i dont want to have to specify the python-version, so CI is blocked by https://github.com/ApeWorX/github-action/pull/33/files

ill release the gh action after that and itll advance us here

**Review by fubuloubu — APPROVED** [review]
In another PR, I was thinking we could also use snekmate a bit for ownership stuff (to have the full 2 phase ownership transfer)

**Comment by antazoey** [comment]
frontend is unable to use the ABIs because of a bug in core Ape, fixed here: https://github.com/ApeWorX/ape/pull/2190

File: sdk/js/index.ts [hunk 1]
```diff
@@ -322,7 +322,7 @@ export default class StreamManager {
     this.publicClient
       .getContractEvents({
         address: this.address,
-        abi: StreamManagerContractType.abi as Abi,
+        abi: StreamManagerABI as ABi,
```
Comment (fubuloubu) [inline sdk/js/index.ts:325]:
```suggestion
        abi: StreamManagerABI as Abi,
```

**Review by fubuloubu — APPROVED** [review]

**Review by fubuloubu — APPROVED** [review]

**Pull request merged**